### PR TITLE
Regression in Machine Learning Tutorial (Version 0.0)

### PR DESCRIPTION
**Pull request by khanteymoori**
Hi all
This is the first version of the "Regression in Machine Learning" tutorial. Please give me your feedback for further improvements.
Best
Alireza

### CODE REVIEW

**Comment by qiagu** [comment]
This looks pretty good. Tool version seems to be a common issue though. IMHO, confusion could be reduced by either 1) marking tool version at right of a tool whenever being mentioned or 2) putting a note in the front of a tutorial, listing all tools and their versions.

**Comment by yvanlebras** [comment]
For the option 2) this can be done directly from a mentionned workflow I guess, and maybe if interest for all tutorials isn't it ? What do you think @bebatut @shiltemann @erasche ?

**Comment by erasche** [comment]
We've had some discussion over time on this issue. There are long term ideas that are promising, but in the short term, we've considered ways to have the tool version information next to the tool name (maybe in a visually unobtrusive way). There is no solution ready today, but if it continues to be an issue for ML, then maybe just write the version directly in the tutorial and we can make it fit some more global standard in the future :)

**Comment by qiagu** [comment]
Thanks @erasche 
@khanteymoori Do you plan to update this tutorial accordingly to the latest v1.0.8.1? 
If needed, I can help a bit.

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+{: .hands_on}
+
+
+> ### {% icon question %} Question
+>
+> Inspect the plots. What can you say about the predictions?
+>
+> > ### {% icon solution %} Solution
+> >
+> > Figures show that the prediction is very good because the predicted age lies close to the true age.
```
Comment (bgruening) [inline topics/statistics/tutorials/regression_machinelearning/tutorial.md]: The Figures

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+>                    - *"Choose a parameter name (with current value)"*: `gradiantboostingregressor_n_estimators: 100`
+>                    - *"Search list"*: `[25, 50, 75, 100, 200]`
+>        - In *"Advanced Options for SearchCV"*:
+>            - *"Select the primary metric (scoring)"*: `Regression -- 'r2'`
+>
+>               A scoring metric can be set. In this tutorial, we use `Regression -- 'r2'`
+>
+>            - *"Select the cv splitter"*: `KFold`
+>
+>               There are different ways to split the dataset into training and validation sets. In our tutorial, we will use `KFold` which splits the dataset into `K` consecutive parts. It is used for cross-validation. It is set to `5` using another parameter `n_splits`.
```
Comment (bgruening) [inline topics/statistics/tutorials/regression_machinelearning/tutorial.md]: consecutive parts for cross-validation. ?

**Comment by simonbray** [comment]
I made a PR with some proposed changes: https://github.com/khanteymoori/training-material/pull/1

The relevant commit is this one: https://github.com/khanteymoori/training-material/pull/1/commits/fe2b51fab8a66b71a33fb35382bc97df9353e683

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+Finally, we will discuss how to train our models by finding the values of their parameters that minimize a cost function. We will work through a real problem to learn how the models and learning algorithms work.
+In this tutorial we will build a regression models to build chronological age prediction based on DNA methylation. In the study [Jana Naue et al. 2017](https://www.sciencedirect.com/science/article/pii/S1872497317301643?via%3Dihub), biomarkers are examined to predict the chronological age of humans by analysing the DNA methylation pattern respectively. Different machine learning algorithms are used in this study to  make age prediction.
+Over the years it is recognized that within each individual, [DNA methylation](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3174260/) changes with age. This knowledge is used to select useful biomarkers from DNA methylation dataset. The [CpGs sites](https://en.wikipedia.org/wiki/CpG_site) with the highest correlation to age are selected as the biomarkers(features). In this tutorial, specific biomarkers are analysed by machine learning algorithms to create an age prediction model.
```
[the remark this answers is not in the document]
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: It is removed

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+>
+> > ### {% icon solution %} Solution
+> >
+> > 75 (Even though the default value of the number of estimators for Gradient boosting regressor is `100`, `75` gives the best accuracy. That's why it is important to perform hyperparameter search to tune these parameters for any dataset). 50 estimators also give almost the same accuracy.
+> >
+> {: .solution}
+>
+{: .question}
+
+Using the **Hyperparameter search** tool, we found the best model, based on the training data. Now, we will predict age in the test dataset using this model.
```
[the remark this answers is not in the document]
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: done

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md (hunk 1 above)
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: done

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md (hunk 2 above)
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: Yes, in cross-validation we need these parts

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+time_estimation: 2H
+contributors:
+- khanteymoori
+- anuprulez
+---
+
+# Introduction
+{:.no_toc}
+
+In this tutorial you will learn how to use galaxy tools in [regression](https://en.wikipedia.org/wiki/Regression_analysis) problems. First, we will introduce regression briefly, and then examine linear regression, which models the relationship between a target variable and some explanatory variables or independet variables. Next, we will discuss gradiant boosting regressor, an advanced regressor model compared to simple linear regression that can model nonlinear relationship between variables. Then, we will show how to visulaize the reults in each step.
```
[the remark this answers is not in the document]
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: done

File: topics/statistics/tutorials/regression_machinelearning/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+
+
+
+# Conclusion
+By following these steps, we learned how to perform regression and visualise the predictions using Galaxy's machine learning and plotting tools. The features of the training dataset are mapped to the real-valued targets. This mapping is used to make predictions on an unseen (test) dataset. The quality of predictions is visualised using a plotting tool. There are multiple other regression algorithms, few are simpler to use (with fewer parameters) and some are powerful, which can be tried out on this dataset and on other datasets as well.
```
[the remark this answers is not in the document]
Comment (khanteymoori) [reply topics/statistics/tutorials/regression_machinelearning/tutorial.md]: is modified

**Comment by khanteymoori** [comment]
The tutorial file is modified.
Thanks @simonbray @bgruening and @qiagu for the feedbacks and comments.

**Comment by bgruening** [comment]
@khanteymoori this looks great to me.
Can you please also add a workflow to this PR?

**Comment by simonbray** [comment]
@khanteymoori asked to go through the tutorial and test it. I did so and made a few changes here: https://github.com/khanteymoori/training-material/pull/2. I also uploaded a clean workflow.

**Comment by bgruening** [comment]
Mh, no clue what this is. I guess I need to run it locally and lint it.

**Pull request merged**